### ISSUE TITLE
Invalid dartdoc references should have visual distinction

### ISSUE THREAD

**Issue by gspencergoog** [flutter/flutter-intellij#3075]
Currently, if there is an invalid dartdoc reference (e.g. [NonExistentClass]) in IntelliJ, the only way to know that it is invalid is to try and command/ctrl-click on the reference and have it say "Cannot find declaration to go to".  It would be super helpful to have those references that are invalid shown in a way that distinguishes them (and configurable in prefs). This would help in avoiding invalid links/typos in the documentation.

**Comment by devoncarew** [comment]
+1

One thing to keep in mind here is that dartdoc understands a broader range of references than the analyzer does. Flutter uses this feature - doc comments there will reference symbols that are not in scope of the current imports (esp., things from further up the stack).

**Comment by gspencergoog** [comment]
Yeah, I know: I had added some imports to make things valid links for the analyzer that I later had to remove (but they're still valid links to dartdoc).  It would be nice if there were a way to specify imports to the analyzer that only apply when processing docs, but that also seems a little complicated and pedantic, so I don't know what the right answer is.
